### ISSUE TITLE
Core 2.2 version won't build

### ISSUE THREAD

**Issue by OnenessUK** [IdentityServer/IdentityServer4#3745]
The Core 2.2 version of Identity Server doesn't currently build out of the box.

I've added global.json:
{
    "sdk": {
      "version": "2.2.402"
    }
  }

but get error: NU1202	Package IdentityServer4.Storage 3.0.0-preview3.4 is not compatible with netstandard2.0 (.NETStandard,Version=v2.0)

This implies there's a dependency in the code base somewhere on  IdentityServer4.Storage 3.0.0-preview3.4 which is obviously inappropriate with .NET Core 2.2.

Or am  I doing something wrong?

**Comment by leastprivilege** [comment]
That would surprise me.

**Comment by OnenessUK** [comment]
It surprised me too.

If you've got  moment Dominick maybe try a fresh download yourself https://github.com/IdentityServer/IdentityServer4/tree/aspnetcore2 and doing a build.

**Comment by leastprivilege** [comment]
i just tried it on a freshly cloned repo. Works fine.

**Comment by OnenessUK** [comment]
Thanks for taking the time to do that.

I just did it again, same result.

Pulled the aspnetcore2 branch, opened up the file IdentityServer4.Core.sln in VS2019 and got the error on Rebuild. See images.

What steps did you take?
![vs](https://user-images.githubusercontent.com/4983027/67108553-b42a4b00-f1c6-11e9-8f38-567dcd0fc062.jpg)g
![git](https://user-images.githubusercontent.com/4983027/67108580-bf7d7680-f1c6-11e9-9726-21bbecc601d8.jpg)

**Comment by leastprivilege** [comment]
Did you run the build script first?

**Comment by echerost** [comment]
Maybe I say nonsense, but can it be related to [this](https://docs.microsoft.com/en-us/ef/core/what-is-new/ef-core-3.0/breaking-changes#netstandard21)?

**Comment by leastprivilege** [comment]
No - you need to run the build script first before you can compile in Visual Studio. 

The readme says that as well.

**Comment by OnenessUK** [comment]
Thanks for your time on this.

The same thing is happening even after running build.ps1.

I download the repo, run Build.ps1, open the solution in VS and error "package IdentityServer4.Storage 3.0.0-preview3.4 is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). Package IdentityServer4.Storage 3.0.0-preview3.4" still occurs.

There is definitely something wrong here with the nuget packages.

Please see attached vs image.

In "Manage Nuget Packages for solution" when I click uninstall on the IdentityServer4.Storage which states version 2.5.4-0 the preview changes box comes up with "Uninstalling: IdentityServer4.Storage 3.0.0-preview3.4". That's a different version so it looks as though there's some corruption of the packages here.

Also, something I've never seen before, in the left panel by IdentityServer4.Storage, it states "not available in this source" ??


![vs2](https://user-images.githubusercontent.com/4983027/67202338-4d907180-f400-11e9-8f89-1f595dbe6484.jpg)

**Comment by leastprivilege** [comment]
You see that there is a nuget.config in the repo. This is important - maybe something overrides that..

**Comment by OnenessUK** [comment]
The content of that file is pretty simple, it just states the location of nuget json and the local nuget folder.

<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
	<clear/>
    <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
    <add key="IdentityServer Local" value="./nuget" />
  </packageSources>
</configuration>

**Comment by leastprivilege** [comment]
I just tried it again. Works for me...

**Comment by OnenessUK** [comment]
Ahh, the common refrain of the developer to the user: "it works fine for me". I'm a developer myself so I know that one.

I've demonstrated that there's an issue with the packages somewhere, I'm doing what should be a very simple process with minimal steps, but as shown above it doesn't work.

It's a shame because we were looking to purchase AdminUI and maybe others, but we'll have to look elsewhere, or maybe we can use the dll out of the box. We'll see.

**Comment by leastprivilege** [comment]
Have you considered that it might be an issue with your machine?

What I tested is this

* clone the repo
* switch to aspnetcore2 branch
* run build.ps1
* open IdentityServer4.sln in VS
* compile 

that's working for me as expected.

And yes - you can use the dll out of the box. That's how it is designed.

**Comment by samirkarki** [comment]
I was also having exactly same issues. I had run build.ps1 but never new there was an issue underlying it. So I run build.ps1 from powershell manually so that I can view the errors while running it. In my case, git path was not in windows variable, and gave me weird errors. I caught it while running clean.ps1. Once I was able to run build.ps1 successfully, I was able to build in VS. I hope this helps @OnenessUK .
